compiler/src/iree/compiler/Codegen/Interfaces/PartitionableLoopsInterface.cpp

```diff
@@ -83,6 +83,23 @@ struct Mmt4DOpPartitionableLoops
   }
 };
 
+/// External model implementation for linalg::BatchMmt4DOp.
+struct BatchMmt4DOpPartitionableLoops
+    : public PartitionableLoopsInterface::ExternalModel<
+          BatchMmt4DOpPartitionableLoops, linalg::BatchMmt4DOp> {
+  llvm::SmallVector<unsigned>
+  getPartitionableLoops(Operation *op,
+                        std::optional<unsigned> maxNumPartitionedLoops) const {
+    SmallVector<unsigned> parallelLoops = {0, 1, 2};
```

Comment:
Can we cast the op to linalg::LinalgOp and use `getNumParallelLoops` method?

Reply:
I think the issue is `batch_mmt4d` has 5 parallel loops but we don't want to partition on the last two. That's why the default implementation doesn't work for it (and for `mmt4d`):

https://github.com/openxla/iree/blob/7d7391f1f989d77d1b4c6b02242f51efee93c084/compiler/src/iree/compiler/Codegen/Interfaces/PartitionableLoopsInterface.cpp#L34-L52

Reply:
oh, good point. sorry for the wrong suggestion, it looks good to me!